OpenRA.Mods.Common/Traits/Capturable.cs

```diff
@@ -32,16 +33,6 @@ public bool CanBeTargetedBy(Actor captor, Player owner)
 			if (c == null)
 				return false;
 
-			var playerRelationship = owner.Stances[captor.Owner];
```

Comment:
You need to check the stances here too (and likewise for the other traits).  The order targeters are client side code, so the checks there could be modified or removed by somebody malicious.